recipes-connectivity/openssh/openssh_%.bbappend

```diff
@@ -1,4 +1,16 @@
 FILESEXTRAPATHS_prepend := "${THISDIR}/files:"
+SAVED_DIR := "${THISDIR}"
+
+do_patch_append() {
+    bb.build.exec_func('do_close_all_ssh_sessions', d)
+}
+
+do_close_all_ssh_sessions () {
+    # Patch sysvinit file to close all open ssh connections on shutdown or reboot
+    #
+    # perform the patch operation on the version of init in the WORKDIR, not the original recipe
+    patch -u ${WORKDIR}/init -i ${SAVED_DIR}/files/close-all-ssh-connections.patch
+}
```

Comment:
What about...
```
do_patch_oe_source () {
    ...
    patch -u ...
}

addtask do_patch_oe_source before do_patch
```

So that it lives as its own task.

Reply:
Ok a bit closer to what you'd ideally like, the patch file still gets sourced from the original location in the recipe since it can't be added to SRC_URI, but now there's a new task after patch do_patch_oe_source, and its cleaner than the do_patch_append implementation.